In [7]:
import pandas as pd
import geopandas as gpd

In [3]:
park_crime=pd.read_csv('../data/1_parkcrime.csv')

In [10]:
shape_path="../data/2 park_location.geojson"
park_shape = gpd.GeoDataFrame.from_file(shape_path).to_crs(epsg=4326)

In [16]:
park_shape.park_name= park_shape.park_name.str.lower()

In [21]:
location = park_shape[['park_name','geometry']]
location = location.rename(columns={'park_name':'PARK'})

In [34]:
score = pd.read_csv('../data/3_park_score.csv')
score = score[['SiteName','average_score']]
score = score.rename(columns={"SiteName":'PARK'})

In [40]:
score.PARK=score.PARK.str.lower()

In [67]:
len(park_crime.PARK.value_counts())

743

#### We have crime record for 743 parks in total.

In [49]:
len(park_crime.merge(score,on='PARK',how='inner')['PARK'].value_counts())

492

#### If we use park score, inner merge, only 492 parks left.

In [42]:
park_crime.merge(score,on='PARK',how='outer')

PERIOD                                  PARK        BOROUGH  \
0     2014-q3                       pelham bay park  BRONX           
1     2014-q4                       pelham bay park  BRONX           
2     2015-q1                       pelham bay park  BRONX           
3     2015-q2                       pelham bay park  BRONX           
4     2015-q3                       pelham bay park  BRONX           
5     2015-q4                       pelham bay park  BRONX           
6     2016-q1                       pelham bay park  BRONX           
7     2016-q2                       pelham bay park  BRONX           
8     2016-q3                       pelham bay park  BRONX           
9     2014-q3                    van cortlandt park  BRONX           
10    2014-q4                    van cortlandt park  BRONX           
11    2015-q1                    van cortlandt park  BRONX           
12    2015-q2                    van cortlandt park  BRONX           
13    2015-q3                    van cortlandt park  BRONX           
14    2015-q4                    van cortlandt park  BRONX           
15    2016-q1                    van cortlandt park  BRONX           
16    2016-q2                    van cortlandt park  BRONX           
17    2016-q3                    van cortlandt park  BRONX           
18    2014-q3          rockaway beach and boardwalk  QUEENS          
19    2014-q4          rockaway beach and boardwalk  QUEENS          
20    2015-q1          rockaway beach and boardwalk  QUEENS          
21    2015-q2          rockaway beach and boardwalk  QUEENS          
22    2015-q3          rockaway beach and boardwalk  QUEENS          
23    2015-q4          rockaway beach and boardwalk  QUEENS          
24    2016-q1          rockaway beach and boardwalk  QUEENS          
25    2016-q2          rockaway beach and boardwalk  QUEENS          
26    2016-q3          rockaway beach and boardwalk  QUEENS          
27    2014-q3                       freshkills park  STATEN ISLAND   
28    2014-q4                       freshkills park  STATEN ISLAND   
29    2015-q1                       freshkills park  STATEN ISLAND   
...       ...                                   ...            ...   
7929      NaN                   saw mill playground            NaN   
7930      NaN                         youth village            NaN   
7931      NaN                    estella diggs park            NaN   
7932      NaN                      grand playground            NaN   
7933      NaN                         barretto park            NaN   
7934      NaN  kingsbridge heights community center            NaN   
7935      NaN                 mount hope playground            NaN   
7936      NaN                  cleopatra playground            NaN   
7937      NaN                  field of dreams park            NaN   
7938      NaN                    galileo playground            NaN   
7939      NaN                   eae j mitchell park            NaN   
7940      NaN                      cpl fischer park            NaN   
7941      NaN                          ranaqua park            NaN   
7942      NaN                      rock garden park            NaN   
7943      NaN                     mount hope garden            NaN   
7944      NaN                         boston garden            NaN   
7945      NaN                  stop & go playground            NaN   
7946      NaN                          seabury park            NaN   
7947      NaN                  beanstalk playground            NaN   
7948      NaN             rev lena irons unity park            NaN   
7949      NaN             ogden plimpton playground            NaN   
7950      NaN                     west farms rapids            NaN   
7951      NaN                           walton park            NaN   
7952      NaN                         morris garden            NaN   
7953      NaN                half-nelson playground            NaN   
7954      NaN    

In [58]:
demo = pd.read_csv('../data/4 cleaned_medianincome_houldhold.csv')
demo = demo[['park_name','household','median_income']]
demo= demo.rename(columns={'park_name':'PARK'})
demo.PARK = demo.PARK.str.lower()

In [65]:
park_crime = park_crime.merge(demo,on='PARK',how = 'inner')

In [73]:
complaints = pd.read_csv('../data/6 311_complains_merged_with_crime.csv')
del complaints['Unnamed: 0']

In [75]:
complaints.merge(demo,on='PARK',how='outer')

PERIOD                                          PARK        BOROUGH  \
0     2014-q3                               pelham bay park  BRONX           
1     2014-q4                               pelham bay park  BRONX           
2     2015-q1                               pelham bay park  BRONX           
3     2015-q2                               pelham bay park  BRONX           
4     2015-q3                               pelham bay park  BRONX           
5     2015-q4                               pelham bay park  BRONX           
6     2016-q1                               pelham bay park  BRONX           
7     2016-q2                               pelham bay park  BRONX           
8     2016-q3                               pelham bay park  BRONX           
9     2014-q3                            van cortlandt park  BRONX           
10    2014-q4                            van cortlandt park  BRONX           
11    2015-q1                            van cortlandt park  BRONX           
12    2015-q2                            van cortlandt park  BRONX           
13    2015-q3                            van cortlandt park  BRONX           
14    2015-q4                            van cortlandt park  BRONX           
15    2016-q1                            van cortlandt park  BRONX           
16    2016-q2                            van cortlandt park  BRONX           
17    2016-q3                            van cortlandt park  BRONX           
18    2014-q3                  rockaway beach and boardwalk  QUEENS          
19    2014-q4                  rockaway beach and boardwalk  QUEENS          
20    2015-q1                  rockaway beach and boardwalk  QUEENS          
21    2015-q2                  rockaway beach and boardwalk  QUEENS          
22    2015-q3                  rockaway beach and boardwalk  QUEENS          
23    2015-q4                  rockaway beach and boardwalk  QUEENS          
24    2016-q1                  rockaway beach and boardwalk  QUEENS          
25    2016-q2                  rockaway beach and boardwalk  QUEENS          
26    2016-q3                  rockaway beach and boardwalk  QUEENS          
27    2014-q3                               freshkills park  STATEN ISLAND   
28    2014-q4                               freshkills park  STATEN ISLAND   
29    2015-q1                               freshkills park  STATEN ISLAND   
...       ...                                           ...            ...   
8657      NaN                colonel charles young triangle            NaN   
8658      NaN                                   fisher pool            NaN   
8659      NaN                      104th st. softball field            NaN   
8660      NaN                                      p.s. 223            NaN   
8661      NaN  saratoga square urban renewal area, site 164            NaN   
8662      NaN                                american heart            NaN   
8663      NaN                              classon triangle            NaN   
8664      NaN                triborough bridge playground a            NaN   
8665      NaN                triborough bridge playground c            NaN   
8666      NaN            west 186th street basketball court            NaN   
8667      NaN                                 lentol garden            NaN   
8668      NaN                triborough bridge playground d            NaN   
8669      NaN                            cutinella triangle            NaN   
8670      NaN                          jamaica gateway park            NaN   
8671      NaN                                       greenst            NaN   
8672      NaN                       merchant's house museum            NaN   
8673      NaN                           garden of happiness            NaN   
8674      NaN                                         mapes            NaN   
8675      NaN                               townsend garden            NaN   
8676      NaN                